analyzer/src/main/kotlin/MavenSupport.kt

```diff
@@ -165,7 +165,7 @@ class MavenSupport(localRepositoryManagerConverter: (LocalRepositoryManager) ->
         val artifactDescriptorRequest = ArtifactDescriptorRequest(artifact, repositories, "project")
         val artifactDescriptorResult = repoSystem
                 .readArtifactDescriptor(repositorySystemSession, artifactDescriptorRequest)
-        val allRepositories = artifactDescriptorResult.repositories + repositories
+        val allRepositories = (artifactDescriptorResult.repositories + repositories).toSet()
```

Comment:
The order of repositories is important, please use `distinct()` instead:
https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/distinct.html

Reply:
Ok, I checked and `toSet()` also keeps the iterativion order:
https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/to-set.html

Reply:
While I had checked, too, that `toSet()` maintains the order, I like your `distinct()` better as it's more explicit and avoids the list -> set -> list conversion. I'll make the change.